### ISSUE TITLE
Error with module lifelines, name 'dmatrices' is not defined

### ISSUE THREAD

**Issue by dlippold** [pgxcentre/genipe#44]
After I installed genipe (on a fully updated Fedora 25 system) as described in the documentation I executed the test and get an error message.

In detail I did the following as root:

`dnf install python3 python3-devel python3-numpy python3-jinja2 python3-pandas python3-setuptools`

And then I did the following as a user:

`pyvenv $HOME/genipe_pyvenv`
`source $HOME/genipe_pyvenv/bin/activate`
`pip install genipe`
`pip install pyfaidx`
`pip install lifelines`
`python`
`import genipe`
`genipe.test()`

After that I get 8 errors with the message `NameError: name 'dmatrices' is not defined`, e.g.:

>ERROR: test_fit_cox_interaction_snp1 (genipe.tests.test_imputed_stats.TestImputedStatsCox)
>Tests the 'fit_cox' function with interaction, first SNP.
>----------------------------------------------------------------------
>Traceback (most recent call last):
>  File "/home/user/genipe_pyvenv/lib64/python3.5/site-packages/genipe/tests/test_imputed_stats.py", line 866, in test_fit_cox_interaction_snp1
>    formula=formula,
>  File "/home/user/genipe_pyvenv/lib64/python3.5/site-packages/genipe/tools/imputed_stats.py", line 1128, in fit_cox
>    y, X = dmatrices(formula, data=data, return_type="dataframe")
>NameError: name 'dmatrices' is not defined

The output of the command `source $HOME/genipe_pyvenv/bin/activate; python --version; deactivate` is: Python 3.5.4

The output of the command `source $HOME/genipe_pyvenv/bin/activate; pip list; deactivate` is:

>cycler (0.10.0)
>genipe (1.3.3)
>Jinja2 (2.10)
>lifelines (0.12.0)
>MarkupSafe (1.0)
>numpy (1.13.3)
>pandas (0.20.0)
>pip (8.1.2)
>pyfaidx (0.5.1)
>pyparsing (2.2.0)
>python-dateutil (2.6.1)
>pytz (2017.3)
>scipy (1.0.0)
>setuptools (25.1.1)
>six (1.11.0)

**Comment by lemieuxl** [comment]
There are multiple modules that are required in order to perform statistical analysis ([requirements section](http://pgxcentre.github.io/genipe/installation.html#requirements) of the documentation).

I think only *patsy* was missing, but you should install all the optional requirements if you aim at using the statistical tool.

The problem is that *lifelines* removed the `normalize` argument since version 0.10 and version 0.9 is not compatible with *pandas* 0.20.

You can either downgrade *lifelines* to version 0.9 and *pandas* to a compatible version (I guess version 0.19 should do the trick).

```console
$ pip list
Package         Version
--------------- -------
biopython       1.70   
cycler          0.10.0 
Cython          0.27.3 
drmaa           0.7.8  
genipe          1.3.3  
Jinja2          2.10   
lifelines       0.9.0.0
MarkupSafe      1.0    
matplotlib      2.1.0  
numpy           1.13.3 
pandas          0.19.0 
patsy           0.4.1  
pip             9.0.1  
pyfaidx         0.5.1  
pyparsing       2.2.0  
python-dateutil 2.6.1  
pytz            2017.3 
scipy           1.0.0  
setuptools      36.2.0 
six             1.11.0 
statsmodels     0.8.0
$ python -c "import genipe; genipe.test()"
...
----------------------------------------------------------------------
Ran 176 tests in 749.269s

OK (skipped=13)
```

I'll try and update *genipe* quickly so that it works with the most current version of all the packages.

**Comment by AmmyDK** [comment]
Excuse me, I have been change genipe from 1.4.2 to 1.3.3 and following all the modules you listed above, but during run "genipe.test()" still many errors. During 1.4.2 version, the running errors are "tie_methods(9errors)" but after downgrading, the errors are "attributeerror(8errors)". How could I correct these warnings?

**Comment by lemieuxl** [comment]
@AmmyDK 
Version 1.5.0 was released and should work with Python 3.7 and above, and with the newest versions of all the dependencies.
